### PR TITLE
[3564][MIG]  product_sales_record: Migration to 15.0

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3564](https://www.quartile.co/web#id=3564&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

**Review by kanda999 — APPROVED** [review]
Function review

File: product_gross_profit_calculation/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            if move.state == "posted" and move.move_type == "out_invoice":
+                for line in move.invoice_line_ids:
+                    template = line.product_id.product_tmpl_id
+                    template.write(
+                        {
+                            "sale_price_unit": line.price_total / line.quantity
+                            if line.quantity
+                            else 0.0,
+                        }
+                    )
```
Comment (yostashiro) [inline product_gross_profit_calculation/models/account_move.py]:
```suggestion
                    line.product_id.write(
                        {
                            "sale_price_unit": line.price_total / line.quantity
                            if line.quantity
                            else 0.0,
                        }
                    )
```

File: product_gross_profit_calculation/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+This module adds some new fields in product template and calculates gross profit of product.
```
Comment (yostashiro) [inline product_gross_profit_calculation/readme/DESCRIPTION.rst:1]:
```suggestion
This module adds some new fields in product template and calculates gross profit of product.

This module currently does not support the case where there are multiple variants per product template.
```

File: product_gross_profit_calculation/tests/test_product_gross_profit_calculation.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                "product_uom_qty": 2.0,
+                "order_id": sale_order.id,
+            }
+        )
+        sale_order.action_confirm()
+        self.assertEqual(self.product.product_tmpl_id.sale_price_unit, 200)
+        self.assertEqual(
+            self.product.product_tmpl_id.confirmation_date, sale_order.date_order
+        )
+        self.assertEqual(self.product.product_tmpl_id.team_id.id, sale_order.team_id.id)
```
Comment (yostashiro) [inline product_gross_profit_calculation/tests/test_product_gross_profit_calculation.py]:
```suggestion
        self.assertEqual(self.product.sale_price_unit, 200)
        self.assertEqual(self.product.confirmation_date, sale_order.date_order)
        self.assertEqual(self.product.team_id.id, sale_order.team_id.id)
```

File: product_gross_profit_calculation/tests/test_product_gross_profit_calculation.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                            "quantity": 2.0,
+                            "account_id": income_account.id,
+                        },
+                    )
+                ],
+            }
+        )
+        # Post invoice and check the sale_price_unit of the product
+        invoice.action_post()
+        self.assertEqual(self.product.product_tmpl_id.sale_price_unit, 400)
```
Comment (yostashiro) [inline product_gross_profit_calculation/tests/test_product_gross_profit_calculation.py]:
```suggestion
        self.assertEqual(self.product.sale_price_unit, 400)
```

**Review by yostashiro — APPROVED** [review]

**Pull request merged**